### ISSUE TITLE
Signal Throw & Catch

### ISSUE THREAD

**Issue by MaheshkumarSundaram** [ralphhanna/bpmn-server#46]
_Using 1.3.2_

### Process 1 (named `RTPTestMsgThrow`):
![image](https://user-images.githubusercontent.com/55505611/187940873-39d27768-253d-4b88-8cda-b1d16f76094b.png)

### Process 2 (named `RTPTestMsgCatch`:
![image](https://user-images.githubusercontent.com/55505611/187940708-9205eba6-7e7e-4969-91ec-8d2449d00040.png)

### Expected Behaviour:

1) Process 1 Starts
2) "Pick Hotel" gets completed
3) Process 1 throws a message for Process 2 to start
4) Process 2 starts
5) Service task "Welcome Guest" gets called
6) Timer wait for 1 min
7) Service task "Checkout" gets called
8) Process 2 ends by signalling an event for Process 1
9) Process 1 picks up the signal
10) Service task "Request Review" gets called
11) Process 1 Ends

### Actual Behaviour:

Until Step 7, the execution is as expected.
On step 8, when the end signal event is launched from process 2, the process `boundary-event` gets called.

![image](https://user-images.githubusercontent.com/55505611/187942733-a4c2ee5b-4ba7-46cc-8042-3dda7c357404.png)

What am I doing wrong & how should I do get my expected behaviour? 

Thanks!

**Comment by ralphhanna** [comment]
can you please include your bpmn files either here or in bpmn.omniworkflow.com

Thanks

**Comment by MaheshkumarSundaram** [comment]
<details>
<summary>RTPTestMsgThrow.bpmn</summary>

```xml

<?xml version="1.0" encoding="UTF-8"?>
<bpmn2:definitions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:bpmn2="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="sample-diagram" targetNamespace="http://bpmn.io/schema/bpmn" xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd">
  <bpmn2:process id="rtp_test_msg_throw" isExecutable="false">
    <bpmn2:sequenceFlow id="Flow_06l4doj" sourceRef="start_test_msg_throw" targetRef="pick_hotel" />
    <bpmn2:sequenceFlow id="Flow_1psk1k8" sourceRef="pick_hotel" targetRef="Event_04d5bzt" />
    <bpmn2:intermediateThrowEvent id="Event_04d5bzt" name="Hotel Stay">
      <bpmn2:incoming>Flow_1psk1k8</bpmn2:incoming>
      <bpmn2:outgoing>Flow_0mnkxzq</bpmn2:outgoing>
      <bpmn2:messageEventDefinition id="MessageEventDefinition_0sqol4b" messageRef="Message_04uwjgl" />
    </bpmn2:intermediateThrowEvent>
    <bpmn2:startEvent id="start_test_msg_throw" name="Start">
      <bpmn2:outgoing>Flow_06l4doj</bpmn2:outgoing>
    </bpmn2:startEvent>
    <bpmn2:userTask id="pick_hotel" name="Pick Hotel">
      <bpmn2:incoming>Flow_06l4doj</bpmn2:incoming>
      <bpmn2:outgoing>Flow_1psk1k8</bpmn2:outgoing>
    </bpmn2:userTask>
    <bpmn2:sequenceFlow id="Flow_0mnkxzq" sourceRef="Event_04d5bzt" targetRef="hotel_stay_end" />
    <bpmn2:sequenceFlow id="Flow_0k2hqc7" sourceRef="hotel_stay_end" targetRef="request_review" />
    <bpmn2:endEvent id="end_test_msg_throw" name="End">
      <bpmn2:incoming>Flow_0o8o9p2</bpmn2:incoming>
    </bpmn2:endEvent>
    <bpmn2:sequenceFlow id="Flow_0o8o9p2" sourceRef="request_review" targetRef="end_test_msg_throw" />
    <bpmn2:serviceTask id="request_review" name="Request Review">
      <bpmn2:incoming>Flow_0k2hqc7</bpmn2:incoming>
      <bpmn2:outgoing>Flow_0o8o9p2</bpmn2:outgoing>
    </bpmn2:serviceTask>
    <bpmn2:intermediateCatchEvent id="hotel_stay_end" name="Stay End">
      <bpmn2:incoming>Flow_0mnkxzq</bpmn2:incoming>
      <bpmn2:outgoing>Flow_0k2hqc7</bpmn2:outgoing>
      <bpmn2:signalEventDefinition id="SignalEventDefinition_0mfanux" signalRef="Signal_1wv5k67" />
    </bpmn2:intermediateCatchEvent>
  </bpmn2:process>
  <bpmn2:message id="Message_04uwjgl" name="hotel_stay_period" />
  <bpmn2:signal id="Signal_1wv5k67" name="confirm_stay_end" />
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="rtp_test_msg_throw">
      <bpmndi:BPMNEdge id="Flow_0o8o9p2_di" bpmnElement="Flow_0o8o9p2">
        <di:waypoint x="840" y="180" />
        <di:waypoint x="902" y="180" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0k2hqc7_di" bpmnElement="Flow_0k2hqc7">
        <di:waypoint x="678" y="180" />
        <di:waypoint x="740" y="180" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0mnkxzq_di" bpmnElement="Flow_0mnkxzq">
        <di:waypoint x="588" y="180" />
        <di:waypoint x="642" y="180" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_1psk1k8_di" bpmnElement="Flow_1psk1k8">
        <di:waypoint x="500" y="180" />
        <di:waypoint x="552" y="180" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_06l4doj_di" bpmnElement="Flow_06l4doj">
        <di:waypoint x="348" y="180" />
        <di:waypoint x="400" y="180" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape id="Event_110yadi_di" bpmnElement="Event_04d5bzt">
        <dc:Bounds x="552" y="162" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="546" y="205" width="52" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_0bxpx3u_di" bpmnElement="start_test_msg_throw">
        <dc:Bounds x="312" y="162" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="318" y="205" width="25" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_1ukt6f4_di" bpmnElement="pick_hotel">
        <dc:Bounds x="400" y="140" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_0toitbo_di" bpmnElement="end_test_msg_throw">
        <dc:Bounds x="902" y="162" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="910" y="205" width="20" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_0c24c4b_di" bpmnElement="request_review">
        <dc:Bounds x="740" y="140" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_1ntlx9v_di" bpmnElement="hotel_stay_end">
        <dc:Bounds x="642" y="162" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="638" y="205" width="45" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn2:definitions>

Please find the BPMN file here:
```
</details>

<details>
<summary>RTPTestMsgCatch.bpmn</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn2:definitions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:bpmn2="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:camunda="http://camunda.org/schema/1.0/bpmn" id="sample-diagram" targetNamespace="http://bpmn.io/schema/bpmn" xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd">
  <bpmn2:process id="rtp_test_msg_catch" isExecutable="false">
    <bpmn2:sequenceFlow id="Flow_059u8la" sourceRef="test_msg_catch_start" targetRef="welcome_guest" />
    <bpmn2:serviceTask id="welcome_guest" name="Welcome Guest" implementation="welcomeGuestService" camunda:delegateExpression="welcomeGuestService">
      <bpmn2:incoming>Flow_059u8la</bpmn2:incoming>
      <bpmn2:outgoing>Flow_08kw2ja</bpmn2:outgoing>
    </bpmn2:serviceTask>
    <bpmn2:sequenceFlow id="Flow_08kw2ja" sourceRef="welcome_guest" targetRef="stay_period" />
    <bpmn2:intermediateCatchEvent id="stay_period" name="Stay">
      <bpmn2:incoming>Flow_08kw2ja</bpmn2:incoming>
      <bpmn2:outgoing>Flow_0wxabew</bpmn2:outgoing>
      <bpmn2:timerEventDefinition id="TimerEventDefinition_0zlqz9d">
        <bpmn2:timeDuration xsi:type="bpmn2:tFormalExpression">PT60S</bpmn2:timeDuration>
      </bpmn2:timerEventDefinition>
    </bpmn2:intermediateCatchEvent>
    <bpmn2:sequenceFlow id="Flow_0wxabew" sourceRef="stay_period" targetRef="check_out" />
    <bpmn2:serviceTask id="check_out" name="Checkout" implementation="checkOutService" camunda:delegateExpression="checkOutService">
      <bpmn2:incoming>Flow_0wxabew</bpmn2:incoming>
      <bpmn2:outgoing>Flow_0zw279a</bpmn2:outgoing>
    </bpmn2:serviceTask>
    <bpmn2:sequenceFlow id="Flow_0zw279a" sourceRef="check_out" targetRef="test_msg_catch_end" />
    <bpmn2:startEvent id="test_msg_catch_start" name="Start">
      <bpmn2:outgoing>Flow_059u8la</bpmn2:outgoing>
      <bpmn2:messageEventDefinition id="MessageEventDefinition_0ltc8ia" messageRef="Message_04uwjgl" />
    </bpmn2:startEvent>
    <bpmn2:endEvent id="test_msg_catch_end" name="End">
      <bpmn2:incoming>Flow_0zw279a</bpmn2:incoming>
      <bpmn2:signalEventDefinition id="SignalEventDefinition_0txkcmd" signalRef="Signal_1wv5k67" />
    </bpmn2:endEvent>
  </bpmn2:process>
  <bpmn2:message id="Message_08pk4qx" name="checkout_guest" />
  <bpmn2:message id="Message_1w7u82r" />
  <bpmn2:message id="Message_04uwjgl" name="hotel_stay_period" />
  <bpmn2:signal id="Signal_1wv5k67" name="confirm_stay_end" />
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="rtp_test_msg_catch">
      <bpmndi:BPMNEdge id="Flow_0zw279a_di" bpmnElement="Flow_0zw279a">
        <di:waypoint x="820" y="-40" />
        <di:waypoint x="972" y="-40" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0wxabew_di" bpmnElement="Flow_0wxabew">
        <di:waypoint x="568" y="-40" />
        <di:waypoint x="720" y="-40" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_08kw2ja_di" bpmnElement="Flow_08kw2ja">
        <di:waypoint x="340" y="-40" />
        <di:waypoint x="532" y="-40" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_059u8la_di" bpmnElement="Flow_059u8la">
        <di:waypoint x="128" y="-40" />
        <di:waypoint x="240" y="-40" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape id="Activity_0v170j3_di" bpmnElement="welcome_guest">
        <dc:Bounds x="240" y="-80" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_1ykgw4q_di" bpmnElement="stay_period">
        <dc:Bounds x="532" y="-58" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="539" y="-15" width="23" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_0biviol_di" bpmnElement="check_out">
        <dc:Bounds x="720" y="-80" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_18t96l8_di" bpmnElement="test_msg_catch_start">
        <dc:Bounds x="92" y="-58" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="98" y="-15" width="25" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_1o5adrx_di" bpmnElement="test_msg_catch_end">
        <dc:Bounds x="972" y="-58" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="980" y="-15" width="20" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn2:definitions>
```
</details>

Thanks!

**Comment by ralphhanna** [comment]
My apologies, there were couple of issues with the code that caused all the confusion.
Your processed worked perfect, once the bugs were fixed.
Thanks for providing a good test case.
![image](https://user-images.githubusercontent.com/11893416/188049125-6caea4b5-1bdd-4a6e-87dd-7107c83e3ee6.png)

![image](https://user-images.githubusercontent.com/11893416/188049280-558fe914-b38f-4bdc-b39d-6b8a0e5a308a.png)

Fix is in 1.3.4

**Comment by MaheshkumarSundaram** [comment]
Thanks for the fix.
